riptide-stream/src/main/java/org/zalando/riptide/stream/StreamSpliterator.java

```diff
@@ -47,6 +47,7 @@ public boolean tryAdvance(Consumer<? super T> action) {
         try {
             JsonToken token = parser.nextToken();
             if (token == null) {
+                parser.close();
```

Comment:
I'd also like to have the close operation on the same logical layer as the creation of the parser (the same applies to io streams), i.e. I'd do the close in the method that creates the parser.